File: 【哈希集合】2023B-找出符合要求的字符串子串.cpp
/*
 * @hw id=2023B lang=cpp
 *
 * 找出符合要求的字符串子串
 */

// @hw code=start
#include<iostream>
#include<set>
#include<algorithm>
using namespace std;

int main() {
    string s1, s2;
    cin >> s1 >> s2;
    set<char> set1(s1.begin(), s1.end());
    set<char> set2(s2.begin(), s2.end());
    string ans;
    for (char c : s1) {
        if (set2.find(c) != set2.end()) {
            ans += c;
        }
    }
    sort(ans.begin(), ans.end());
    cout << ans << endl;
    return 0;
}
// @hw code=end

File: 【哈希集合】2023B-找出符合要求的字符串子串.java
/*
 * @hw id=2023B lang=java
 *
 * 找出符合要求的字符串子串
 */

// @hw code=start
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        String s1 = in.nextLine();
        String s2 = in.nextLine();
        Set<Character> set1 = new HashSet<>();
        for (char c : s1.toCharArray()) {
            set1.add(c);
        }
        Set<Character> set2 = new HashSet<>();
        for (char c : s2.toCharArray()) {
            set2.add(c);
        }
        StringBuilder sb = new StringBuilder();
        for (char c : set1) {
            if (set2.contains(c)) {
                sb.append(c);
            }
        }
        char[] ans = sb.toString().toCharArray();
        Arrays.sort(ans);
        System.out.println(new String(ans));
    }
}
// @hw code=end